src/foam/nanos/analytics/ruler/LogAnalyticEventRuleAction.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.analytics.ruler',
+  name: 'LogAnalyticEventRuleAction',
+
+  implements: [
+    'foam.nanos.ruler.RuleAction'
+  ],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.analytics.AnalyticEvent',
+    'foam.nanos.session.Session',
+    'java.util.Date'
+  ],
+
+  properties: [
+    {
+      name: 'eventName',
+      class: 'String'
+    },
+    {
+      name: 'traceIdExpr',
+      class: 'foam.mlang.ExprProperty'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        agency.submit(x, new ContextAgent() {
+          @Override
+          public void execute(X x) {
+            var event = new AnalyticEvent();
+            event.setName(getEventName());
+            event.setTimestamp(new Date());
```

Comment:
Is this an after rule?  Thinking if beneficial if the timestamp aligns with the objects created or lastmodified timestamp.

Reply:
This is a rule action can be on either before or after rule. Would like to use for after rule in https://github.com/nanoPayinc/NANOPAY/pull/15547 so the id will be generated and can be used as the traceId of the analytic event.

We can consider using object created/lastModified as the timestamp but the difference would just be some milliseconds and we will need to handle if the object doesn't have created or lastModified based on the rule operation.

Reply:
we'll lets do that: if lastModified > 0 else created

Reply:
Forget my last comment.  the timestamp as is fine.